gobblin-core/src/main/java/gobblin/writer/http/AsyncHttpWriterBuilder.java

```diff
@@ -88,6 +115,8 @@ protected void validate() {
         .checkNotNull(getResponseHandler(), "ResponseHandler is required for " + this.getClass().getSimpleName());
   }
 
+  protected abstract String getLimiterKey (State state) throws NotConfiguredException;
```

Comment:
Since we don't know where the limit key may come from, we can omit the parameter `state`, which can be directly retrieved by `getState`